packages/core/app-layout/src/components/navbar/AccountDropdown.vue

```diff
@@ -0,0 +1,72 @@
+<template>
+  <KDropdownMenu
+    appearance="selectionMenu"
+    class="top-bar-dropdown-menu"
+    :disabled="disabled"
+    :disabled-tooltip="tooltip"
+    :kpop-attributes="{ placement: 'bottomStart', disabled: disabled }"
+    :label="string"
+    :width="width"
+    @change="(selection: any) => optionSelected(selection)"
+  >
+    <template #items="{ handleSelection }">
+      <slot>
+        <KDropdownItem
+          v-for="option in options"
+          :key="option.value"
+          :data-testid="'select-' + option.value"
+          :selected="selectedOption.value === option.value"
+          @click="handleSelection(option)"
+        >
+          <slot :option="option">
+            {{ option.label }}
+          </slot>
+        </KDropdownItem>
+      </slot>
+    </template>
+  </KDropdownMenu>
+</template>
+
+<script setup lang="ts">
+import { ref, Ref, PropType } from 'vue'
+import type { NavbarDropdownMenuItem } from '../../types'
+
+const props = defineProps({
+  options: {
+    type: Array as PropType<NavbarDropdownMenuItem[]>,
```

Comment:
Reusing the `NavbarDropdownMenuItem ` type here since it provides the needed type